plugins/bundle/plugin.go

```diff
@@ -257,16 +257,25 @@ func (p *Plugin) Loaders() map[string]Loader {
 
 // Trigger triggers a bundle download on all configured bundles.
 func (p *Plugin) Trigger(ctx context.Context) error {
+	var errs Errors
+
 	p.mtx.Lock()
 	downloaders := map[string]Loader{}
 	for name, dl := range p.downloaders {
 		downloaders[name] = dl
 	}
 	p.mtx.Unlock()
 
-	for _, d := range downloaders {
-		// plugin callback will log the trigger error and include it in the bundle status
-		_ = d.Trigger(ctx)
+	for name, d := range downloaders {
+		// plugin callback will also log the trigger error and include it in the bundle status
+		err := d.Trigger(ctx)
+		// only return errors for TriggerMode manual as periodic bundles will be retried
+		if err != nil && *p.config.Bundles[name].Trigger == plugins.TriggerManual {
```

Comment:
Use the `Config()` method on the plugin to retrieve the config.

Reply:
@ashutosh-narkar do you think it makes sense to only return the errors for bundles in `manual` trigger mode?
It felt sensible to me as `periodic` triggermode will be retried by the bundle plugin but might also just create confusion.
it also creates this additional if statement which will then not be tested.

Reply:
Manual trigger mode can currently only be configured via a library integration. So to keep parity with the discovery plugin, returning errors in the manual mode makes sense.